dao-contracts/src/voting/governance_voting.rs

```diff
@@ -0,0 +1,366 @@
+//! Governance Voting module.
+pub mod consts;
+pub mod events;
+pub mod voting;
+
+use casper_dao_utils::bytes::BytesConversion;
+
+use casper_dao_utils::{
+    casper_contract::unwrap_or_revert::UnwrapOrRevert,
+    casper_dao_macros::Instance,
+    casper_env::{call_contract, emit, get_block_time, revert, self_address},
+    Address, Error, Mapping, Variable,
+};
+use casper_types::{runtime_args, RuntimeArgs, U256, U512};
+
+use crate::{
+    ReputationContractCaller, ReputationContractInterface, VariableRepositoryContractCaller,
+};
+
+use self::{
+    events::{
+        FormalVotingEnded, InformalVotingEnded, VoteCast, VotingContractCreated, VotingCreated,
+    },
+    voting::{Voting, VotingConfiguration, VotingResult, VotingType},
+};
+
+use casper_dao_utils::consts as dao_consts;
+
+use super::{vote::VotingId, Vote};
+
+/// The Governance Voting module.
+
+#[derive(Instance)]
+pub struct GovernanceVoting {
+    pub variable_repo: Variable<Option<Address>>,
```

Comment:
Make those fields private. This module will be reused multiple times, so it's better to have clear functions-based interface.